### PR TITLE
Added a keyboard shortcut (command key + enter key) to create a new Note entry

### LINKED ISSUE #6532: Notes Form Entry
### As ...

Ozzie - Operator

### I want ...

to have the Notes form entry accept the key combination of CMD + Enter to hit the submit button.

### So that ...

The app performs like many form entry systems on the web.

### I know this is done when...

- I type in a new note
- Hit CMD + Enter that the "Create" Button gets selected.

### Optional - Feature groups this request pertains to.

- [ ] Automation
- [ ] Circuits
- [ ] DCIM
- [ ] IPAM
- [ ] Misc (including Data Sources)
- [ ] Organization
- [ ] Apps (and other Extensibility)
- [ ] Security (Secrets, etc)
- [ ] Image Management
- [X] UI/UX
- [ ] Documentation
- [ ] Other (not directly a platform feature)

### Database Changes

None

### External Dependencies

_No response_

### PR DESCRIPTION
**Pull request by HanlinMiao**
# Closes #6532 
# What's Changed

Added a keyboard shortcut (command key + enter key) to create a new Note entry
# Screenshots

N/A
# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example App Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/core/templates/generic/object_notes.html [hunk 1]
```diff
@@ -31,3 +31,16 @@ <h3>Add a new note</h3>
     {% include 'responsive_table.html' %}
     {% include 'inc/paginator.html' with paginator=table.paginator page=table.page %}
 {% endblock %}
+
+{% block javascript %}
+    {{ block.super }}
+    <script>
+        document.addEventListener('keydown', function(event) {
+        // command + enter
+        if (event.metaKey && event.key === 'Enter') {
```
[the remark this answers is not in the document]
Comment (smk4664) [reply nautobot/core/templates/generic/object_notes.html:40]: How about we add both the metaKey and the ctrlKey in the check?

[the remark this answers is not in the document]
Comment (gsnider2195) [reply nautobot/core/templates/generic/object_notes.html:40]: I think that's the appropriate thing to do here

[the remark this answers is not in the document]
Comment (gsnider2195) [reply nautobot/core/templates/generic/object_notes.html]: We should limit this event to the textarea and maybe apply it to all textareas in forms?

```
function submitOnEnter(event) {
    if ((event.metaKey || event.ctrlKey) && event.key === 'Enter') {
        if (!event.repeat) {
            const newEvent = new Event("submit", {cancelable: true});
            event.target.form.dispatchEvent(newEvent);
        }

        event.preventDefault(); // Prevents the addition of a new line in the text field
    }
}

document.querySelectorAll("textarea.form-control").forEach(function(element) {element.addEventListener("keydown", submitOnEnter)});
```

File: changes/6532.added [hunk 2]
```diff
@@ -0,0 +1 @@
+Added a keyboard shortcut (command key + enter key) to create a new Note entry.
```
Comment (gsnider2195) [inline changes/6532.added:1]:
```suggestion
Added a keyboard shortcut (⌘+enter or ctrl+enter) to submit forms when typing in a textarea.
```

**Comment by gsnider2195** [comment]
This is working for me in firefox but not chrome. I'll see if I can figure out why

File: nautobot/project-static/js/forms.js [hunk 3]
```diff
@@ -931,10 +931,21 @@ function createInput(element){
     replaceEl(element, input_field)
 }
 
+function submitOnEnter(event) {
+    if ((event.metaKey || event.ctrlKey) && event.key === 'Enter') {
+        if (!event.repeat) {
+            const newEvent = new Event("submit", {cancelable: true});
+            event.target.form.dispatchEvent(newEvent);
+        }
```
[the remark this answers is not in the document]
Comment (gsnider2195) [reply nautobot/project-static/js/forms.js:939]: https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/requestSubmit

[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/project-static/js/forms.js:939]: Thanks for the investigation on this!

**Review by gsnider2195 — APPROVED** [review]

**Pull request merged**